sys/net/bpf.c

```diff
@@ -125,7 +125,7 @@ struct bpf_program_buffer {
 #ifdef BPF_JITTER
 	bpf_jit_filter		*func;
 #endif
-	void			*buffer[];
+	void			*buffer[0] __no_subobject_bounds;
```

Comment:
This looks wrong?

Reply:
I originally dropped it after your comment in the last patch but reintroduced it because we have an assertion in `containerof` that checks for this and to remove it requires compiler changes in order to get the same compile-time assertion.

Reply:
Why are we doing `containerof` on this?

Reply:
Now changed to store the `bpf_program_buffer` directly into the `bpf_d`.